### ISSUE TITLE
Introducing the Model Optimization Toolkit for TensorFlow

### ISSUE THREAD

**Issue by pangafu** [leela-zero/leela-zero#2030]
https://medium.com/tensorflow/introducing-the-model-optimization-toolkit-for-tensorflow-254aca1ba0a3?linkId=57036398

Benefits of post-training quantization
4x reduction in model sizes
Models, which consist primarily of convolutional layers, get 10–50% faster execution
RNN-based models get up to 3x speed-up
Due to reduced memory and computation requirements, we expect that most models will also have lower power consumption

So how to optimization leela zero model to make it smaller and faster?

**Comment by alreadydone** [comment]
@Ttl already implemented INT8 inference in his cudnn branches, but they only support 192x15 nets last time I checked. I think they are able to load any 192x15 net and perform quantization/calibration in real time, but maybe I misunderstand.

**Comment by gcp** [comment]
>So how to optimization leela zero model to make it smaller and faster?

The main problem is that the hardware needs support for <fp32 compute (or in NVIDIA's case, expose it in the driver). The examples on that page are all with phones for that reason.

Right now only GTX 1080 Ti and RTX cards would be able to benefit from this, IF the accuracy loss is not prohibitive.

(We already do "fp16" quantization which AMD cards support but that doesn't require any special calibration)

**Comment by Ttl** [comment]
Int8 inference is about 2x faster than float on GTX10xx GPUs, but in my experiments just directly quantitizing a network resulted in big loss of strength even with dynamic range calibration. Int8 network needs to be retrained with quantization operations inserted in the Tensorflow but this retrained model is now much weaker with float inference. So in practice int8 needs its own network which only works with int8 inference. This makes it not very interesting from the training point of view.

On GTX1050 TI the int8 inference is faster than single precision OpenCL at batch size 1 and the retrained int8 net was stronger at equal time. On GTX1080 Ti int8 net was slower at batch size 1 and needs batching before it's useful for playing.

**Comment by sethtroisi** [comment]
closing old announcement issue with no open action items.
